src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContextProvider.cs

```diff
@@ -39,7 +40,7 @@ public async Task<AggregateCrossTargetProjectContext> CreateProjectContextAsync(
             ImmutableDictionary<string, ConfiguredProject> configuredProjectsMap = await _activeConfiguredProjectsProvider.GetActiveConfiguredProjectsMapAsync();
 #pragma warning restore CS0618 // Type or member is obsolete
             ProjectConfiguration activeProjectConfiguration = _commonServices.ActiveConfiguredProject.ProjectConfiguration;
-            ImmutableArray<ITargetFramework>.Builder targetFrameworks = ImmutableArray.CreateBuilder<ITargetFramework>(initialCapacity: configuredProjectsMap.Count);
+            var targetFrameworks = PooledArray<ITargetFramework>.GetInstance();
```

Comment:
What's the impact of losing the capacity specifier here?

Reply:
In cases where we know _exactly_ what the array size will be upon construction, we do not want to use pooled builders. It's better to allocate the array size exactly then move it to the returned `ImmutableArray<>`. A pooled builder performs a defensive copy, in addition to potentially resizing during construction.

On that basis, the three removals of `MoveToImmutable` are all regressions.

Reply:
@drewnoakes 	did some prefview trace'n and confirmed that memory and cpu time is sub-optimal for all cases if we do not use `MoveToImmutable`.  I'll revert this change for those cases